codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -857,9 +876,15 @@ public Http2Stream closeRemoteSide() {
         }
 
         @Override
-        public int nextStreamId() {
-            // For manually created client-side streams, 1 is reserved for HTTP upgrade, so start at 3.
-            return nextStreamId > 1 ? nextStreamId : nextStreamId + 2;
+        public int incrementAndGetNextStreamId() {
+            return nextReservationStreamId >= 0 ? nextReservationStreamId += 2 : nextReservationStreamId;
+        }
+
+        private void incrementExpectedStreamId(int streamId) {
+            if (streamId > nextReservationStreamId && nextReservationStreamId >= 0) {
```

Comment:
I removed `isExhausted` because it was not clear which is exhausted (the reserved streams or the "created streams") and seemed more clear at the time to directly use the variables so that ambiguity didn't exist.

However I can add something like `isReservedStreamIdsExhausted` if that improves things...WDYT?